src/toil/batchSystems/mesos/batchSystem.py

```diff
@@ -90,7 +90,7 @@ def __init__(self, config, maxCores, maxMemory, maxDisk, masterAddress):
         self.jobQueues = JobQueue()
 
         # Address of the Mesos master in the form host:port where host can be an IP or a hostname
-        self.masterAddress = masterAddress
+        self.masterAddress = config.masterAddress
```

Comment:
Thanks for looking at it - much appreciated. Python is awesome. I'll push up that change and if it still fails I'll wait for #1738.